python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -639,6 +639,21 @@ def test_read_directory_s3fs(s3_example_s3fs, use_legacy_dataset):
     assert result.equals(table)
 
 
+@pytest.mark.pandas
+@parametrize_legacy_dataset
+def test_read_single_file_list(tempdir, use_legacy_dataset):
+    base_path = str(tempdir)
+    data_path = os.path.join(base_path, 'data.parquet')
```

Comment:
```suggestion
    data_path = str(base_path / 'data.parquet')
```

The `tempdir` fixture is a pathlib object, so that allows to write it like the above

Reply:
Thanks! I did not know you could do that. I have added a commit with a change similar to the suggested `data_path = str(base_path / 'data.parquet')`